pkg/services/gitopswriter/writer_add_test.go

```diff
@@ -0,0 +1,265 @@
+	"github.com/weaveworks/weave-gitops/pkg/testutils"
+)
+
+var (
+	ctx context.Context
+)
+
+func createRepoURL(url string) gitproviders.RepoURL {
+	repoURL, err := gitproviders.NewRepoURL(url)
+	Expect(err).NotTo(HaveOccurred())
+
+	return repoURL
+}
+
+func createDirWriter() GitOpsDirectoryWriter {
+	repoWriter := gitrepo.NewRepoWriter(app.ConfigURL, gitProviders, gitClient, log)
+	automationSvc := automation.NewAutomationService(gitProviders, fluxClient, log)
+
+	return NewGitOpsDirectoryWriter(automationSvc, repoWriter, osysClient, log)
+}
+
+var dummyGitSource = []byte(`---
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+  name: wego-fork-test
+  namespace: wego-system
+spec:
+  interval: 30s
+  ref:
+    branch: main
+  secretRef:
+    name: wego-test-cluster-wego-fork-test
+  url: ssh://git@github.com/user/wego-fork-test.git`)
+
+var _ = Describe("Add", func() {
+	var _ = BeforeEach(func() {
+		app = models.Application{
+			Name:                "bar",
+			Namespace:           wego.DefaultNamespace,
+			HelmSourceURL:       "",
+			GitSourceURL:        createRepoURL("ssh://git@github.com/foo/bar.git"),
+			Branch:              "main",
+			Path:                "./kustomize",
+			AutomationType:      models.AutomationTypeKustomize,
+			SourceType:          models.SourceTypeGit,
+			HelmTargetNamespace: "",
+		}
+
+		gitProviders.GetDefaultBranchReturns("main", nil)
+
+		ctx = context.Background()
+	})
+
+	Context("add app with config in app repo", func() {
+		BeforeEach(func() {
+			app.GitSourceURL = createRepoURL("ssh://git@github.com/foo/bar.git")
+			app.ConfigURL = app.GitSourceURL
+			gitOpsDirWriter = createDirWriter()
+
+			gitClient.OpenStub = func(s string) (*gogit.Repository, error) {
+				r, err := gogit.Init(memory.NewStorage(), memfs.New())
+				Expect(err).ShouldNot(HaveOccurred())
+
+				_, err = r.CreateRemote(&config.RemoteConfig{
+					Name: "origin",
+					URLs: []string{"git@github.com:foo/bar.git"},
+				})
+				Expect(err).ShouldNot(HaveOccurred())
+				return r, nil
+			}
+		})
+
+		Describe("generates application goat", func() {
+			It("clones the repo to a temp dir", func() {
+				err := gitOpsDirWriter.AddApplication(ctx, app, "test-cluster", true)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(gitClient.CloneCallCount()).To(Equal(1))
+				_, repoDir, url, branch := gitClient.CloneArgsForCall(0)
+
+				Expect(repoDir).To(ContainSubstring("user-repo-"))
+				Expect(url).To(Equal("ssh://git@github.com/foo/bar.git"))
+				Expect(branch).To(Equal("main"))
+			})
+
+			It("writes the files to the disk", func() {
+				fluxClient.CreateSourceGitReturns(dummyGitSource, nil)
+				fluxClient.CreateKustomizationReturns([]byte("kustomization"), nil)
+
+				err := gitOpsDirWriter.AddApplication(ctx, app, "test-cluster", true)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(gitClient.WriteCallCount()).To(Equal(5))
+
+				path, content := gitClient.WriteArgsForCall(0)
```

Comment:
I see that two of the files aren't being checked (only the original three are) but if they were, wouldn't it be okay to check that the git client writes: <data> to <path> rather than having it write through to an actual git repo?

Reply:
Assuming these are existing tests, we don't need to fix it here. Blocker removed for me.

To answer the question: since the `It` block says "writes the files to disk", we should be able to check their contents. Unless this is going over the network?

Reply:
On second look, are these existing tests that were moved @jrryjcksn ?